app/frontend/src/features/communities/PagePage.tsx

```diff
@@ -0,0 +1,61 @@
+import PageTitle from "../../components/PageTitle";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { useAppDispatch } from "../../store";
+import { Page } from "../../pb/pages_pb"
+import { service } from "../../service";
+import Alert from "../../components/Alert";
+import CircularProgress from "../../components/CircularProgress";
+import TextBody from "../../components/TextBody";
+import { useHistory } from "react-router-dom";
+import { pageRoute } from "../../AppRoutes"
+import Markdown from "../../components/Markdown";
+
+export default function PagePage() {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+  const [page, setPage] = useState(null as Page.AsObject | null);
+
+  const dispatch = useAppDispatch();
+  const history = useHistory();
+
+  const { pageId, pageSlug } = useParams<{ pageId: string, pageSlug?: string }>();
+
+  useEffect(() => {
+    if (!pageId) return;
+    (async () => {
+      setLoading(true);
+      try {
+        const page = await service.pages.getPage(Number(pageId))
+        setPage(page);
+        if (page.slug !== pageSlug) {
+          // if the address is wrong, redirect to the right place
+          history.push(`${pageRoute}/${page.pageId}/${page.slug}`);
+        }
+      } catch (e) {
+        setError(e.message);
+      }
+      setLoading(false);
+    })();
+  }, [pageId]);
+
```

Comment:
I think `useQuery` is preferred.

Reply:
I don't know what I'm doing, could you elaborate, please?

Reply:
I haven't reviewed yet but you can do something like:

```
const {isLoading, isError, error, data: page} = useQuery<Page.AsObject, Error>(
  ["page", pageId], //query key
  () => service.pages.getPage(+pageId), //query fn
)
//put a useEffect here for the redirect
```